### PR TITLE
Make systemd unit more production-like

### PR DESCRIPTION
**Pull request by tiandrey**
**Type: feature**

Fixed the following things in systemd unit:

* Added `Restart=always` to ensure verdaccio keeps running after fail caused by plugins (e.g. verdaccio-ldap, see https://github.com/Alexandre-io/verdaccio-ldap/issues/39)
* Removed totally useless `ExecStop` clause
* Added `--config` start parameter

### CODE REVIEW

**Comment by juanpicado** [comment]
Please read errors

File: systemd/verdaccio.service [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-Description=verdaccio Service
+Description=Verdaccio lightweight npm proxy registry
 
 [Service]
 Type=simple
+Restart=always
 User=verdaccio
-WorkingDirectory=/home/verdaccio
-ExecStart=/usr/bin/env verdaccio
-ExecStop=/usr/bin/bash -c "kill $(ps -ef | grep ^verdacc+ | awk {'print $2'})"
```
Comment (juanpicado) [inline systemd/verdaccio.service]: Why did you remove `ExecStop`?

File: systemd/verdaccio.service [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+Group=verdaccio
+ExecStart=/usr/bin/verdaccio --config /etc/verdaccio/config.yaml
```
Comment (juanpicado) [inline systemd/verdaccio.service]: The default conf is not there by default, how the user knows about this location?

File: systemd/verdaccio.service (hunk 1 above)
Comment (tiandrey) [reply systemd/verdaccio.service]: Because
1) systemd automatically kills all processes spawned by service (read `man 5 systemd.service`) and
2) you command line kills _all_ processes of user `verdaccio` (or whichever user gets abbreviated by ps to `verdacc+`).
Also, in such situations use `pkill` instead of `kill $(ps | grep)` - it's much more convenient.

File: systemd/verdaccio.service (hunk 2 above)
Comment (tiandrey) [reply systemd/verdaccio.service]: `/home/verdaccio` is not there by default neither :)
The cause of this pull request is to make systemd unit production-ready, e.g. make it usable for system-wide installation on production servers (as opposed to development or home servers).
As FHS (Filesystem Hierarchy Standard, `man 7 hier`) states, variable state information for programs (e.g. data) should be stored in `/var/lib/` (e.g. `/var/lib/verdaccio` in your case), while `/home/` is for home directories of users (human users, not system). Moreover, system users usually do not have home directories (of course they have some directory specified as their home directory, but no "personal" files are stored there). Thus using `/home/verdaccio` for data storage is not correct in terms of FHS and that's why I removed `WorkingDirectory=/home/verdaccio`.
As for configuration files, FHS states that they should be stored in `/etc` (so in your case that would be `/etc/verdaccio.yaml` or `/etc/verdaccio/*.yaml`). None of example configuration files would work out of the box on production (`default.yaml` - "It allows all users to do anything, so don't use it on production systems.", `docker.yaml` - uses non-standard paths, `full.yaml` - has a lot of placeholders to change), so anyway user would have to create his own copy of configuration file, so I think `/etc/verdaccio/config.yaml` would be an apparent name for a arbitrary system-wide config for Verdaccio.

**Comment by tiandrey** [comment]
I updated commit, and my big comment to your second question vanished from conversation :( I hope you still have it in your mailbox.

File: systemd/verdaccio.service (hunk 2 above)
Comment (dlouzan) [reply systemd/verdaccio.service]: Do we have some kind of package build automation? (e.g. deb or rpm) Or is anybody downstream building and packaging this for linux distributions?

In any case, I agree with @tiandrey about the comments, if daemonizing verdaccio, as a sysadmin that's where I would expect to find the configuration files and runtime state data.

Comment (tiandrey) [reply systemd/verdaccio.service]: I personally use fpm (https://github.com/jordansissel/fpm) - it builds DEBs and RPMs from npm just fine. And it automatically installs configs from `conf` subdir to `/etc/verdaccio`.

**Comment by dlouzan** [comment]
@tiandrey Ooops, yes, for some reason it also disappeared for me :-/ I don't see your comment anywhere. I'm adding my comment again (luckily I use a paste manager xD)

File: systemd/verdaccio.service (hunk 1 above)
Comment (dlouzan) [reply systemd/verdaccio.service]: Do we have some kind of package build automation? (e.g. deb or rpm) Or is anybody downstream building and packaging this for linux distributions?

In any case, I agree with @tiandrey about the comments, if daemonizing verdaccio, as a sysadmin that's where I would expect to find the configuration files and runtime state data.

Comment (dlouzan) [reply systemd/verdaccio.service]: Any reason for not using `Restart=on-failure`? I haven't used verdaccio as a daemon in a normal system, but the signal handling with dumb-init seems to be ok in the docker build.

Regarding `ExecStop`, I also don't see the need for it in principle, systemd should keep track of the spawned process id.

**Comment by dlouzan** [comment]
Your comment email just came in, I'm pasting

**Comment by juanpicado** [comment]
@tiandrey could you add a new section here https://github.com/verdaccio/verdaccio/blob/master/docs/server.md and document all these new things? There was no info before, but I'm sure user will ask in the future.

**Comment by tiandrey** [comment]
Sure. Updated docs.

**Comment by dlouzan** [comment]
It looks good to me, except the pending question about the `Restart` option.
